### PR TITLE
Fix provisioning script hang

### PR DESCRIPTION
**Pull request by amstewart**
Author: Alex Stewart <alex.stewart@ni.com>
Date:   Wed Mar 17 14:04:18 2021 -0500

    initrd: fixup /init script
    
    When commit 7455c348dce8f52e12907f7862ec49307169737f "improved the
    payload handling", it also renamed the $NIRECOVERY_MOUNT variable to
    $NIRECOVERY_MOUNTPOINT. One branch structure did not get updated and so
    broke detection for USB provisioning. Further, the branch structure uses
    single-quotes around the variable, which causes bash to literally
    interpret the variable name, instead of substituting its value.
    
    Signed-off-by: Alex Stewart <alex.stewart@ni.com>

----

@ni/rtos @chaitu236 @billpittman 

@chaitu236 Can you try this PR on your machine and see if the provisioning script still hangs?

### CODE REVIEW

**Review by shruthi-ravi — APPROVED** [review]

**Comment by amstewart** [comment]
tldr; see [this commit](https://github.com/ni/meta-nilrt/pull/142/commits/dbbf5ae4091eb7ec2ec98e5e97354ce452f66ffe). The rest of this PR is just reverting the revert of @billpittman's patchset.

**Review by jeminor — APPROVED** [review]

**Comment by chaitu236** [comment]
@amstewart I haven't been able to validate this on my machine yet due to the issue I mentioned on Teams. But I don't want to gate this PR as I'm sure this is the fix. So I'm going to approve it.

**Review by chaitu236 — APPROVED** [review]

**Review by billpittman — APPROVED** [review]
